### DISCUSSION TITLE
Not seeing sensors in NMEA 2k network

### DISCUSSION THREAD

**Discussion by jonasbager** [hatlabs/discussions#116] [General]
Hi there. I have connected my sensors and uploaded code in sync with the tutorial provided. However, No sensors show up on the NMEA network. I look in the debug code and see that it says:

(load_configuration)(C1) Loading configuration for path '/System/WiFi Settings' from '/1y6oPD6lHzs1o19UpJFFjbslZu8='
(load_configuration)(C1) Could not find configuration for path /System/Signal K Settings
(load_configuration)(C1) Not loading configuration: no config_path specified: 
(load_configuration)(C1) Could not find configuration for path /mainEngineOilTemp/oneWire
(OneWireTemperature)(C1) FATAL: Unable to allocate a OneWire sensor for /mainEngineOilTemp/oneWire. All sensors have already been configured. Check the physical wiring of your sensors.
(load_configuration)(C1) Could not find configuration for path /mainEngineCoolantTemp/oneWire
(OneWireTemperature)(C1) FATAL: Unable to allocate a OneWire sensor for /mainEngineCoolantTemp/oneWire. All sensors have already been configured. Check the physical wiring of your sensors.
(load_configuration)(C1) Could not find configuration for path /mainEngineWetExhaustTemp/oneWire
(OneWireTemperature)(C1) FATAL: Unable to allocate a OneWire sensor for /mainEngineWetExhaustTemp/oneWire. All sensors have already been configured. Check the physical wiring of your sensors.
(load_configuration)(C1) Could not find configuration for path /mainEngineOilTemp/skPath
(load_configuration)(C1) Could not find configuration for path /mainEngineCoolantTemp/skPath
(load_configuration)(C1) Could not find configuration for path /mainEngineTemp/skPath
(load_configuration)(C1) Could not find configuration for path /mainEngineWetExhaustTemp/skPath
(start)(C1) Starting subsystems

It seems that it failes to find the configurations, but I am just not sure what I am doing wrong?

**Comment by mairas** [comment]
There's probably something wrong with your hardware setup because none of your OneWire sensors could be found.

**Comment by jonasbager** [comment]
Hmm. I guess so. But I connected one of the sensors directly to the 3v, gnd and pin 4, to test if that was the case

**Reply by mairas** [reply]
I believe you're referring to SH-ESP32, then? The reserved GPIO header pins (the ones with a circle around them) on SH-ESP32 aren't connected by default. You need to connect the respective solder jumper on the bottom of the board if you want to use them.

**Comment by jonasbager** [comment]
Ah okay. Did not understand that at all from the tutorial. So you're saying that I need to connect it like this (see image) or just on any of the pin-4 line on the proto board?
![SH-ESP32-r2-render-e1634637627263](https://github.com/hatlabs/discussions/assets/2243646/10c88ef0-4a55-4b69-9f83-5f20b889f763)


Do I also need to do that with gnd and 3v?

**Reply by mairas** [reply]
The 1-Wire header is at the bottom edge of the board, labeled "1Wire". You need to connect all three pins: DQ, 3V3, and GND.

**Comment by jonasbager** [comment]
Ah of course. This one:
But do I also need to solder those pins to something or are they already connected?
(sorry if all of this is in the docs, but I have really tried to find it in there without success) 

![SH-ESP32-r2-render-e1634637627263](https://github.com/hatlabs/discussions/assets/2243646/b1e85146-cf3a-4f80-9c7f-96b56d5bdd4d)

**Reply by mairas** [reply]
The connector pins are internally connected, yes. Have a look at this page for a description on SH-ESP32 peripherals and connectors:

https://docs.hatlabs.fi/sh-esp32/pages/hardware/

**Comment by jonasbager** [comment]
Thanks! It seems that I now get the connection :)  Thanks a lot!!

However, I still don't see them on my NMEA network, and I get the error that it can't find the configuration files:

--- Terminal on /dev/cu.wchusbserial120 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
(SetupSerialDebug)(C1) 
Serial debugging enabled
(load_configuration)(C1) Loading configuration for path '/System/WiFi Settings' from '/1y6oPD6lHzs1o19UpJFFjbslZu8='
(load_configuration)(C1) Could not find configuration for path /System/Signal K Settings
(load_configuration)(C1) Not loading configuration: no config_path specified: 
(DallasTemperatureSensors)(C1) Found OneWire sensor 28:5c:1f:43:d4:43:3d:3c
(DallasTemperatureSensors)(C1) Found OneWire sensor 28:d3:93:57:04:49:3c:8e
(DallasTemperatureSensors)(C1) Found OneWire sensor 28:1b:f8:43:d4:2f:7d:16
(load_configuration)(C1) Could not find configuration for path /mainEngineOilTemp/oneWire
(OneWireTemperature)(C1) Registered a new OneWire sensor
(load_configuration)(C1) Could not find configuration for path /mainEngineCoolantTemp/oneWire
(OneWireTemperature)(C1) Registered a new OneWire sensor
(load_configuration)(C1) Could not find configuration for path /mainEngineWetExhaustTemp/oneWire
(OneWireTemperature)(C1) Registered a new OneWire sensor
(load_configuration)(C1) Could not find configuration for path /mainEngineOilTemp/skPath
(load_configuration)(C1) Could not find configuration for path /mainEngineCoolantTemp/skPath
(load_configuration)(C1) Could not find configuration for path /mainEngineTemp/skPath
(load_configuration)(C1) Could not find configuration for path /mainEng

**Reply by mairas** [reply]
Those errors are normal. The configuration files are created only when the respective values are saved in the web UI. Until then, default values are used.
